### PR TITLE
Fix pointed out typos

### PR DESCRIPTION
**Pull request by jakobod**
This PR fixes two typos that @andrismednis pointed out in #974 .

### CODE REVIEW

**Comment by andrismednis** [comment]
Here is another case with "relase":
In libcaf_core/caf/actor_storage.hpp there is a template
```
template <class T>
void intrusive_ptr_release(actor_storage<T>* x) {
  // release implicit weak pointer if the last strong ref expires
  // and destroy the data block
  if (x->ctrl.strong_refs.fetch_sub(1, std::memory_order_acq_rel) == 1) {
    x->destroy_data();
    intrusive_ptr_relase_weak(x);    <---- relase !?
  }
}
```
In 6 other places it is 'intrusive_ptr_release_weak'.

File: libcaf_core/caf/actor_storage.hpp [hunk 1]
```diff
@@ -132,7 +132,7 @@ void intrusive_ptr_release(actor_storage<T>* x) {
   // and destroy the data block
   if (x->ctrl.strong_refs.fetch_sub(1, std::memory_order_acq_rel) == 1) {
     x->destroy_data();
-    intrusive_ptr_relase_weak(x);
+    intrusive_ptr_release_weak(x);
```
Comment (Neverlord) [inline libcaf_core/caf/actor_storage.hpp]: This is the only place where we've misspelled `intrusive_ptr_release_weak`, which means this should've resulted in a compiler error. But it didn't. Dead code?

Seems like it when looking at the [coverage report](https://jenkins.inet.haw-hamburg.de/job/CAF/job/actor-framework/job/master/378/cobertura/caf_test/libcaf_core_caf_actor_storage_hpp/). Can you check whether we can just remove all the `intrusive_ptr_` functions from `actor_storage.hpp`? As far as I can tell, we never form an `intrusive_ptr` to an actor storage. Only to an actor control block.

Comment (jakobod) [reply libcaf_core/caf/actor_storage.hpp]: I will check that tomorrow.

Comment (jakobod) [reply libcaf_core/caf/actor_storage.hpp]: Done.

**Pull request merged**